admin/src/main/resources/io/buoyant/admin/js/router_client.js

```diff
@@ -83,17 +83,47 @@ var RouterClient = (function() {
     return summary;
   }
 
-  return function (metricsCollector, routers, client, $metricsEl, routerName, clientTemplate, metricPartial, $chartEl, colors) {
+  return function (metricsCollector, routers, client, $metricsEl, routerName, clientTemplate, metricPartial, $chartEl, colors, $toggleLinks, shouldExpandInitially) {
     template = clientTemplate;
     Handlebars.registerPartial('metricPartial', metricPartial);
     var clientColor = colors.color;
     var latencyLegend = createLatencyLegend(colors.colorFamily);
     var metricDefinitions = getMetricDefinitions(routerName, client.label);
 
+    var $expandLink = $toggleLinks.find(".client-expand");
+    var $collapseLink = $toggleLinks.find(".client-collapse");
+
     renderMetrics($metricsEl, client, [], [], clientColor);
-    var chart = SuccessRateGraph($chartEl, colors.color);
+    var chart = SuccessRateGraph($chartEl.find($(".client-success-rate")), colors.color);
+
+    // collapse client section by default (deal with large # of clients)
+    if(shouldExpandInitially) {
+      showClient();
+    } else {
+      toggleClientDisplay(false);
+    }
+
+    $expandLink.click(showClient);
+    $collapseLink.click(hideClient);
+
+    function showClient() {
+      metricsCollector.registerListener(metricsHandler, getDesiredMetrics);
+      toggleClientDisplay(true);
+    }
 
-    var metricsHandler = function(data) {
+    function hideClient() {
+      metricsCollector.deregisterListener(metricsHandler);
+      toggleClientDisplay(false);
+    }
+
+    function toggleClientDisplay(expand) {
+      $metricsEl.toggle(expand);
+      $chartEl.toggle(expand);
+      $collapseLink.toggle(expand);
+      $expandLink.toggle(!expand);
+    }
```

Comment:
tioli: consider moving the register/deregister calls into `toggleClientDisplay` and removing show/hideClient methods.

if this could cause handlers to be registered more than once, we could modify `registerListener` to check for existence before adding.

if that causes `registerListener` to be expensive, maybe we could change its underlying implementation to a hash?

realizing this may be a lot of work as i'm typing this, just some things to think about.